pkg/helmreconciler/reconciler.go

```diff
@@ -141,26 +141,43 @@ func (h *HelmReconciler) processRecursive(manifests ChartManifestsMap) *v1alpha2
 				<-s
 				log.Infof("Dependency for %s has completed, proceeding.", c)
 			}
+
+			// Set status when reconciling starts
+			status := v1alpha2.InstallStatus_RECONCILING
 			mu.Lock()
 			if _, ok := out.Status[c]; !ok {
 				out.Status[c] = &v1alpha2.InstallStatus_VersionStatus{}
+				out.Status[c].Status = status
 			}
 			mu.Unlock()
-			status := v1alpha2.InstallStatus_RECONCILING
+
+			// Process manifests and get the status result
 			errString := ""
-			if len(m) != 0 {
+			if len(m) == 0 {
+				status = v1alpha2.InstallStatus_NONE
+			} else {
 				status = v1alpha2.InstallStatus_HEALTHY
-				if err := h.ProcessManifest(m[0]); err != nil {
+				if cnt, err := h.ProcessManifest(m[0]); err != nil {
 					errString = err.Error()
 					status = v1alpha2.InstallStatus_ERROR
+				} else if cnt == 0 {
+					status = v1alpha2.InstallStatus_NONE
 				}
 			}
-			mu.Lock()
-			out.Status[c].Status = status
-			if errString != "" {
-				out.Status[c].Error = errString
+
+			// Update status based on the result
+			if status == v1alpha2.InstallStatus_NONE {
+				mu.Lock()
```

Comment:
nit: this is such a short section you may as well move mu.Lock/Unlock outside the if statement to avoid duplication.

Reply:
In this case (all branches of if..else will write `out`), you are right. I will update.